ot/bregman.py

```diff
@@ -455,13 +476,19 @@ def sinkhorn_knopp(a, b, M, reg, numItermax=1000,
             if log:
                 log['err'].append(err)
 
+            if err < stopThr:
+                break
             if verbose:
-                if cpt % 200 == 0:
+                if ii % 200 == 0:
                     print(
                         '{:5s}|{:12s}'.format('It.', 'Err') + '\n' + '-' * 19)
-                print('{:5d}|{:8e}|'.format(cpt, err))
-        cpt = cpt + 1
+                print('{:5d}|{:8e}|'.format(ii, err))
+    else:
+        warnings.warn("Sinkhorn did not converge. You might want to "
```

Comment:
print warning only if verbose ?

Reply:
i mean it never conveges for small egularizaytion and you don't want warning at each iteration in your gradient descent ;)